sample/Benchmark/EscapeTagNames.cs

```diff
@@ -0,0 +1,73 @@
+using System.IO;
+using BenchmarkDotNet.Attributes;
+
+namespace Benchmark
+{
+    [MemoryDiagnoser]
+    public class EscapeTagNames
+    {
+        private const int N = 500;
+
+        [Params("my_tag", "my tag")]
+        public string TagName { get; set; }
+
+        [Benchmark(Baseline = true)]
+        public string NoEscaping()
+        {
+            var writer = new StringWriter();
+
+            for (var i = 0; i < N; i++)
+            {
+                writer.Write(TagName);
+            }
+
+            return writer.ToString();
+        }
+
+        [Benchmark]
+        public string Replace()
+        {
+            var writer = new StringWriter();
+
+            for (var i = 0; i < N; i++)
+            {
+                writer.Write(TagName
+                    .Replace("=", "\\=")
+                    .Replace(" ", "\\ ")
+                    .Replace(",", "\\,"));
+            }
+
+            return writer.ToString();
+        }
+
+        [Benchmark]
+        public string WriteCharOrEscapeString()
```

Comment:
Perhaps the benchmark could call the actual implementation here, so that we can track `LineProtocolWriter.WriteEscapedName()`? Making it internal and adding an `[InternalsVisibleTo()]` attribute would do the trick.